paparazzi/src/main/java/app/cash/paparazzi/internal/VideoUtils.kt

```diff
@@ -0,0 +1,222 @@
+      // after the first bad frame
+      var encoder: AWTSequenceEncoder? =
+        if (!sameFrameCount || !sameDuration) AWTSequenceEncoder.createSequenceEncoder(output, fps) else null
+      var badFrameCount = 0
+      val goldenDefaultImage = goldenFrameGrab.mediaInfo.createBlankFrame()
+      val testDefaultImage = testFrameGrab.mediaInfo.createBlankFrame()
+      for (index in 0 until max(testMeta.totalFrames, goldenMeta.totalFrames)) {
+        val goldenImage = goldenFrameGrab.nextFrame(goldenDefaultImage)
+        val testImage = testFrameGrab.nextFrame(testDefaultImage)
+        val goldenFrameIndex = goldenFrameGrab.videoTrack.curFrame
+        val testFrameIndex = testFrameGrab.videoTrack.curFrame
+
+        val result = calculateDifference(goldenImage, testImage)
+        if (result.percentDifferent > maxPercentDifferent) {
+          if (encoder == null) {
+            //Lazy init delta encoder after we've found a bad frame
+            encoder = AWTSequenceEncoder.createSequenceEncoder(output, fps).also {
+              //Encode the deltas up to the current frame
+              goldenFrameGrab.seekToFramePrecise(0)
+              testFrameGrab.seekToFramePrecise(0)
+              for (i in 0 until index) {
+                val goldenImage2 = goldenFrameGrab.nextFrame(goldenDefaultImage)
+                val testImage2 = testFrameGrab.nextFrame(testDefaultImage)
+
+                val result = calculateDifference(goldenImage2, testImage2)
+                it.encodeImage(result.deltaImage)
+              }
+
+              //Skip current frame since we already have it
+              goldenFrameGrab.nextFrame(goldenDefaultImage)
+              testFrameGrab.nextFrame(testDefaultImage)
+              assert(goldenFrameIndex == goldenFrameGrab.videoTrack.curFrame)
+              assert(testFrameIndex == testFrameGrab.videoTrack.curFrame)
+            }
+          }
+
+          badFrameCount++
+        }
+
+        encoder?.encodeImage(result.deltaImage)
+      }
+      encoder?.finish()
+
+      var error = ""
+      if (badFrameCount > 0) {
+        error += String.format("$badFrameCount video frames differed by more than %.1f%%\n", maxPercentDifferent)
+      }
+
+      if (!sameDuration) {
+        error += "Mismatched video duration expected: ${goldenMeta.totalDuration} actual: ${testMeta.totalDuration}\n"
+      }
+
+      if (!sameFrameCount) {
+        error += "Mismatched frame count expected: ${goldenMeta.totalFrames} actual: ${testMeta.totalFrames}\n"
+      }
+
+      if (error.isNotEmpty()) {
+        Assert.fail(error + "find delta video at ${failureDir.absoluteFile}")
+      }
+    } finally {
+      NIOUtils.closeQuietly(goldenChannel)
+      NIOUtils.closeQuietly(testChannel)
+    }
+  }
+
+  @Throws(IOException::class)
+  private fun calculateDifference(
+    goldenImage: BufferedImage,
+    actualImage: BufferedImage
+  ) : Result {
+    var goldenImage = goldenImage
```

Comment:
Nit: name shadowing

Reply:
I need help on how to login cashapp on my device

Reply:
Hi @8373838, this is not the correct place for those questions. Please head to https://cash.app/help and our Support team should be able to assist you.